### PR TITLE
Add require-await lint rule

### LINKED ISSUE #4394: lint: add "require-await" rule to ESLint config
https://eslint.org/docs/rules/require-await

> There is an eslint rule of [`require-await`](https://eslint.org/docs/rules/require-await) which would catch this as well. With TypeScript checking the return type it is very unlikely enforcing this would cause incorrect behaviour.

@kitsonk I did a quick check and we'd have to add a lot of supressions to tests (because there's a lot of mocks that assert async functions were called). I'm open to that change, but I'd open a new issue for that.

_Originally posted by @bartlomieju in https://github.com/denoland/deno/pull/4384#issuecomment-599469694_

A new rule "require-await" should be added to `.eslintrc.json`; then code in `core/`, `cli/` and `std/` should be linted accordingly.

### PR DESCRIPTION
**Pull request by samrith-s**
Added `require-await` in linting rule to fix #4394.

### CODE REVIEW

File: cli/js/permissions.ts [hunk 1]
```diff
@@ -47,17 +47,17 @@ export class PermissionStatus {
...
-  async revoke(desc: PermissionDescriptor): Promise<PermissionStatus> {
+  revoke(desc: PermissionDescriptor): PermissionStatus {
     const state = permissionsOps.revoke(desc);
     return new PermissionStatus(state);
   }
 
-  async request(desc: PermissionDescriptor): Promise<PermissionStatus> {
+  request(desc: PermissionDescriptor): PermissionStatus {
     const state = permissionsOps.request(desc);
     return new PermissionStatus(state);
```
Comment (nayeemrmn) [inline cli/js/permissions.ts]: Wrap the return types here in `Promise.resolve()`.

Comment (samrith-s) [reply cli/js/permissions.ts]: Gotcha. Thanks!

**Comment by kitsonk** [comment]
@samrith-s so far so good!

**Comment by bartlomieju** [comment]
@samrith-s thank you for this cumbersome work! I fixed some tests and reverted a few changes that weren't necessary.

**Comment by samrith-s** [comment]
> @samrith-s thank you for this cumbersome work! I fixed some tests and reverted a few changes that weren't necessary. 

You’re welcome! I got stuck trying to fix these errors. Thank you so much for stepping in.

**Pull request merged**